### PR TITLE
Implement 'Set EKF Origin' in fly view

### PR DESCRIPTION
**Pull request by Williangalvani**
This is useful for ardupilot for navigation using odometry and other gps-less modes.
set_ekf_origin

Is this relevant for PX4? should it be hidden behind a `supportsSettingEkfOrigin()` in FirmwarePlugin?

This needs testing. My sensors are dead...

### CODE REVIEW

**Review by patrickelectric — APPROVED** [review]

**Comment by bkueng** [comment]
PX4 implements this as well. However from a UI perspective I'm not sure where it makes sense to expose this to all users.

**Comment by dagar** [comment]
FYI @mcsauder

**Comment by dagar** [comment]
How about slightly more generic naming as the message (SET_GPS_GLOBAL_ORIGIN) isn't EKF specific and it isn't necessarily an EKF on the other side?

http://mavlink.io/en/messages/common.html#SET_GPS_GLOBAL_ORIGIN

**Comment by mrpollo** [comment]
@bkueng @dagar do you have time to help me push this through the finish line? its a good feature that could benefit users

**Review by mcsauder — APPROVED** [review]
Looks great.  Sorry it took so long.  Thanks for your work!

**Comment by mcsauder** [comment]
Fix up the conflicts and maybe rebase.  After that it should be good to go!

-Mark

**Comment by mrpollo** [comment]
Thanks @mcsauder I went ahead and fixed the conflicts so we can get this in ASAP. I would like to make a new point release soon.

**Review by bkueng — COMMENTED** [review]
- At the minimum please change ekf to estimator
- Can you only expose this if no gps is used? It's in SYS_STATUS.

**Comment by patrickelectric** [comment]
ping @Williangalvani can you update the code based on the review and rebase over master ?

**Comment by DonLakeFlyer** [comment]
I can clean this up. But is there some way to understand when this command is necessary? As far as I can tell from the current state of the code showSetEkfOrigin is always true. So every QGC user is going to see this new menu item. Even though a massive percentage of users will have no idea what it's used for.

**Comment by Williangalvani** [comment]
With Ardupilot, this is useful when you are flying a vehicle with no absolute positioning. In our case, we have ROVs with very accurate DVL (think velocity sensors) which can be used for navigation. But first the user has to manually tell the vehicle where it is for it to have a starting point.

**Comment by DonLakeFlyer** [comment]
Yeah I get that part. But I"m assuming this is only needed in certain situations. For example: Should this option only be shown if QGC isn't receiving any GPS telemetry? OR something else like that?

**Comment by Williangalvani** [comment]
Right, from Ardupilot's perspective. we should be able to show this only if we have GLOBAL_POSITION_INT showing lat/long == (0,0)
There might be a better option, let me ask the other devs

**Comment by DonLakeFlyer** [comment]
> * Can you only expose this if no gps is used? It's in SYS_STATUS.

Oh, wait I just saw this. Does this work for PX4 and ArduPilot?

**Comment by Williangalvani** [comment]
Yes that makes sense! it is actually how I hid the "No GPS Fix" message for sub in here:
https://github.com/mavlink/qgroundcontrol/pull/9701/files#diff-86909c8722a0a27e4936cfbe08a7e0b463c6f8100575adf84375e312659e9508R558

File: src/FlightDisplay/GuidedActionsController.qml [hunk 1]
```diff
@@ -127,6 +130,7 @@ Item {
     property bool showROI:              _guidedActionsEnabled && _vehicleFlying && __roiSupported && !_missionActive
     property bool showLandAbort:        _guidedActionsEnabled && _vehicleFlying && _fixedWingOnApproach
     property bool showGotoLocation:     _guidedActionsEnabled && _vehicleFlying
+    property bool showSetEkfOrigin:     true
```
Comment (DonLakeFlyer) [inline src/FlightDisplay/GuidedActionsController.qml:133]: So then this should change to: 
```
property bool showSetEkfOrigin:     _activeVehicle.sensorsPresentBits & 32 // MAV_SYS_STATUS_SENSOR_GPS
```

**Comment by DonLakeFlyer** [comment]
@Williangalvani ?

**Pull request closed without merging**